src/credentials/sso/cache.ts

```diff
@@ -31,32 +32,40 @@ export interface SsoCache {
 
 const cacheDir = join(homedir(), '.aws', 'sso', 'cache')
 
-export function getCache(directory = cacheDir): SsoCache {
+export function getCache(directory = cacheDir, statFunc = getFileStats): SsoCache {
+    try {
+        deleteOldFiles(directory, statFunc)
+    } catch (e) {
+        getLogger().warn('Error deleting old files in cache', e)
+    }
+
     return {
         token: getTokenCache(directory),
         registration: getRegistrationCache(directory),
     }
 }
 
-export function getRegistrationCache(directory = cacheDir): KeyedCache<ClientRegistration, RegistrationKey> {
-    const hashScopes = (scopes: string[]) => {
-        const shasum = crypto.createHash('sha256')
-        scopes.forEach(s => shasum.update(s))
-        return shasum.digest('hex')
-    }
-
-    const getTarget = (key: RegistrationKey) => {
-        const suffix = `${key.region}${key.scopes && key.scopes.length > 0 ? `-${hashScopes(key.scopes)}` : ''}`
-        return join(directory, `aws-toolkit-vscode-client-id-${suffix}.json`)
-    }
+function deleteOldFiles(directory: string, statFunc: typeof getFileStats) {
+    const fileNames = readdirSync(directory)
+    fileNames.forEach(fileName => {
+        const filePath = join(directory, fileName)
+        if (!isOldFile(filePath, statFunc)) {
```

Comment:
When deleting files in a loop it's worth being paranoid. If somehow `/` or `.` is passed as `directory`, or any other weird case, it would be catastrophic. 

Some guards that can prevent catastrophe:

- check that `directory` is absolute
- check that the trimmed result of `directory` is non-empty and has length greater than, say, 5